crypto/src/main/java/net/corda/v5/crypto/CompositeKey.java

```diff
@@ -19,30 +22,42 @@
  * Composite key implementations will track the minimum total weight required (in the simple case – the minimum number of child
  * signatures required) to satisfy the subtree rooted at this node.
  */
-interface CompositeKey: PublicKey {
+public interface CompositeKey extends PublicKey {
+
+
+    /**
+     * Constant specifying the maximum number of children keys in the [CompositeKey].
+     */
+    int COMPOSITE_KEY_CHILDREN_LIMIT = 10;
```

Comment:
It was in CryptoUtils.kt before, so yeah, it's different bytecode, but this seemed as good a way to handle this as I could find.

Reply:
Does it need to be in API?
I.e. will we ever want to change it without breaking the API?
I admit I don't know much about how this is used, but this feels more like an implementation constraint.

Maybe @mnesbit can opine?

Reply:
Let's move to `corda-runtime-os`. The implementation of `CompositeKey` is in corda-runtime-os. The CorDapp might call the factory function `CompositeKeyGeneration,.create` with too many keys, in which the factory function will  throw `IllegalArgumentFunction`. We can put the limit in some documentation if we want, and increase it in further version.

(Personally I'd have done this in a separate PR before GA but since others seem keen to do some API cleanup now then so be it).

Reply:
Thanks.

The main reason I think this is important is that this leaves the option of somebody referencing this value in their CorDapp, so taking it away later means that change will break the CorDapp.

So it's in a slightly different category compared to things we can fix up without impacting the CorDapp API.

Reply:
Removed in my local work.

Reply:
pushed